core/thread/auth.go

```diff
@@ -0,0 +1,155 @@
+package thread
+
+import (
+	"context"
+	"encoding"
+	"encoding/base64"
+	"fmt"
+	"strings"
+
+	"github.com/gogo/status"
+	"github.com/grpc-ecosystem/go-grpc-middleware/util/metautils"
+	"github.com/libp2p/go-libp2p-core/crypto"
+	"google.golang.org/grpc/codes"
+)
+
+type Identity interface {
+	Sign([]byte) ([]byte, error)
+	GetPublic() crypto.PubKey
+}
+
+type Auth struct {
+	pk  crypto.PubKey
+	sig []byte
+}
+
+type Method byte
```

Comment:
`Method` provides a tight scoping of the message being signed, since some API methods take the same args.